dialoptions.go

```diff
@@ -635,3 +636,17 @@ func WithResolvers(rs ...resolver.Builder) DialOption {
 		o.resolvers = append(o.resolvers, rs...)
 	})
 }
+
+// WithUseBytesPoolForParser returns a DialOption that specifies whether to use
+// a bytes pool for parsing. Setting this to true will reduce the memory allocation
+// in the parser.
+//
+// # Experimental
+//
+// Notice: This API is EXPERIMENTAL and may be changed or removed in a
+// later release.
+func WithUseBytesPoolForParser(useBytesPoolForParser bool) DialOption {
```

Comment:
I think we should bikeshed a little on this name. Things that come to my mind are:
- WithSharedRecvBuffers
- WithRecvBuffersFromSharedPool
- WithRecvBuffersFromSyncPool

Thoughts @dfawley 

Also, the parameter `useBytesPoolForParser` could be renamed to `ok` since the purpose of the dial option should be clear from the name of it, and this parameter is simply setting the dial option `ON` or `OFF`.

Reply:
Same as above.